File: Application/StarSystems.Core/Models/Star.cs
using System.Collections.Generic;

namespace StarSystems.Core.Models
{
    public partial class Star
    {
        public Star()
        {
            Planet = new HashSet<Planet>();
        }

        public int StarId { get; set; }
        public string StarName { get; set; }
        public string SpectralType { get; set; }
        public double? BvColorIndex { get; set; }
        public double? AbsoluteMagnitude { get; set; }
        public double? Mass { get; set; }
        public double? Radius { get; set; }
        public double? Luminosity { get; set; }
        public int? Temperature { get; set; }
        public double? Age { get; set; }
        public double? Metallicity { get; set; }
        public int? DiscoveryYear { get; set; }
        public int StarSystemId { get; set; }

        public StarSystem StarSystem { get; set; }
        public ICollection<Planet> Planet { get; set; }
    }
}

File: Application/StarSystems/client/src/components/star.js
import React from "react"
import Planet from "./planet"

// @jsx jsx
import { jsx } from 'theme-ui'

import SystemVisualization from "./systemVisualization/systemVisualization"

const Star = ({ star }) => (
  <div>
    <h2>
      {star.starName}
    </h2>
    {star.planets.length > 0 ? <SystemVisualization star={ star } /> : ``}
    <div
      sx = {{
        display: `grid`,
        gridTemplateColumns: `auto 1fr`,
        gridAutoRows: `1.75rem`,
        gridGap: `0 20px`,
        borderBottom: `3px solid #777`
      }}
    >
      <strong>Spectral Classification:</strong>
      <span>{star.spectralType}</span>

      <strong>Absolute Magnitude:</strong>
      <span>{star.absoluteMagnitude ? star.absoluteMagnitude.toFixed(3) : `0.000`}</span>

      <strong>Mass:</strong>
      <span>{star.mass ? star.mass.toFixed(3) : ``} M<sub>Sun</sub></span>

      <strong>Radius:</strong>
      <span>{star.radius ? star.radius.toFixed(3) : ``} R<sub>Sun</sub></span>

      <strong>Luminosity:</strong>
      <span>{star.luminosity ? star.luminosity.toFixed(3) : ``} L<sub>Sun</sub></span>

      <strong>Temperature:</strong>
      <span>{star.temperature} K</span>

      <strong>Year Discovered:</strong>
      <span>{star.discoveryYear}</span>
    </div>
    <div
      sx={{
        marginLeft: 20
      }}
    >
      {star.planets
        .sort((a, b) => a.orbit.semiMajorAxis < b.orbit.semiMajorAxis ? -1 : 1)
        .map(planet =>
        <Planet key={planet.id} planet={planet}></Planet>
      )}
    </div>
  </div>
)

export default Star